python/ray/util/ray_executor.py

```diff
@@ -0,0 +1,237 @@
+import itertools
+import time
+from concurrent.futures import Executor
+from functools import partial
+
+import ray
+from ray.util.annotations import PublicAPI
+
+
+@PublicAPI(stability="alpha")
+class RayExecutor(Executor):
+    """`RayExecutor` is a drop-in replacement for `ProcessPoolExecutor` and
+    `ThreadPoolExecutor` from `concurrent.futures` but distributes and executes
+    the specified tasks over a Ray cluster instead of multiple processes or
+    threads.
+    """
+
+    def __init__(self, max_workers=None, **kwargs):
+        """Initialises a new RayExecutor instance which distributes tasks over
+        a Ray cluster.
+
+        Args:
+            max_workers:
+                The maximum number of Ray workers to distribute the
+                tasks over. This is achieved by creating a fixed pool
+                of Actors (see
+                https://docs.ray.io/en/latest/ray-core/actors/
+                actor-utils.html#actor-pool).
+
+            All additional keyword arguments are passed to ray.init()
+            (see https://docs.ray.io/en/latest/ray-core/package-ref.html#ray-init).
+
+            For example, this will connect to a cluster at the specified address:
+            .. code-block:: python
+
+                RayExecutor(address='192.168.0.123:25001')
+
+            Note: excluding an address will initialise a local Ray cluster.
+        """
+        self._shutdown_lock = False
+        self._futures = []
+        self.__actor_pool = None
+        self.__remote_fn = None
+        self.context = None
+
+        """
+        The following is necessary because `@ray.remote` is only available at runtime.
```

Comment:
<img width="858" alt="Screen Shot 2023-01-18 at 8 29 53 PM" src="https://user-images.githubusercontent.com/18510752/213355416-ceea9e2f-2be3-4b61-a58d-ba2ae7300218.png">
this is incorrectly rendered to the documentation. Can you just use a regular comment 

```
# The following is necessary because `@ray.remote` is only available at runtime.
```